tests/Unit/Repositories/ProfileRepositoryTest.php

```diff
@@ -254,4 +254,38 @@ public function profiles_should_be_grouped()
             $this->assertTrue(array_key_exists($item['display_name'], $profiles['profiles']));
         });
     }
+
+    /**
+     * @covers App\Repositories\ProfileRepository::getProfilesByGroupOrderPiped
+     * @test
+     */
+    public function profile_group_ids_should_return_ordered_array()
+    {
+        // Mock the user listing
+        $return_user_listing = app('Factories\Profile')->create(10);
+
+        $groups = collect($return_user_listing)->map(function ($item) {
+            return array_shift($item['groups']);
+        })->unique()->toArray();
+
+        krsort($groups);
+
+        $piped_groups = implode('|', array_keys($groups));
```

Comment:
This line is interesting. I don't quite understand it.

Reply:
Since the whole point of the "getProfilesByGroupOrderPiped" is to take piped in group ids, we have to turn the available group ids from the faker data into the same format it'll be working with: '0|1|2|3'